src/main/java/training/chessington/model/pieces/Knight.java

```diff
@@ -15,6 +16,40 @@ public Knight(PlayerColour colour) {
 
     @Override
     public List<Move> getAllowedMoves(Coordinates from, Board board) {
-        return new ArrayList<>();
+        List<Move> allowableMoves = new ArrayList<>();
+
+        // laying rectangle
+        getCornersOfRectangle(from, board, allowableMoves, 1, 2);
+        // standing rectangle
+        getCornersOfRectangle(from, board, allowableMoves, 2, 1);
+
+        return allowableMoves;
+    }
+
+    private void getCornersOfRectangle(Coordinates from, Board board,List<Move> initAllowableMoves, int rowInc, int colInc) {
+        int fromRow = from.getRow();
+        int fromCol = from.getCol();
+
+        int[] rowArray = {fromRow - rowInc, fromRow + rowInc};
+        int[] colArray = {fromCol - colInc, fromCol + colInc};
+
+        for(int r: rowArray) {
+
+            // quick exit to next item
+            if (r < 0 | r > 7) {
+                continue;
+            }
+
+            for (int c: colArray) {
+                // quick exit to next item
+                if (c < 0 | c > 7) {

```

Comment:
There's not much benefit to using `r` and `c` as names, it's fine to use full words to make the code more readable :D

You can also avoid needing to use the `continue` statements by just inverting your if statements i.e. 
```
if (r >= 0 && r <= 8) {
    for (int c: colArray) {
        if (c >= 0 && c <= 8) {
            //your code here
        }
    }
}
```